kinto/core/views/heartbeat.py

```diff
@@ -1,20 +1,37 @@
+from concurrent.futures import ThreadPoolExecutor, TimeoutError
 from pyramid.security import NO_PERMISSION_REQUIRED
 
+from kinto import logger
 from kinto.core import Service
 
+
 heartbeat = Service(name="heartbeat", path='/__heartbeat__',
                     description="Server health")
 
 
 @heartbeat.get(permission=NO_PERMISSION_REQUIRED)
 def get_heartbeat(request):
     """Return information about server health."""
-    status = {}
 
+    # Start executing heartbeats concurrently.
     heartbeats = request.registry.heartbeats
+    pool = ThreadPoolExecutor(max_workers=len(heartbeats.keys()))
+    futures = []
     for name, callable in heartbeats.items():
-        status[name] = callable(request)
+        futures.append((name, pool.submit(callable, request)))
 
+    # Wait the results, with timeout.
+    seconds = float(request.registry.settings['heartbeat_timeout_seconds'])
+    status = {}
+    for name, future in futures:
+        try:
+            status[name] = future.result(timeout=seconds)
```

Comment:
Hum, I don't think so. The callable starts running when it is submitted to the pool, that's why I have two loops!